flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java

```diff
@@ -882,42 +784,28 @@ protected ClusterClient retrieveClient(CommandLineOptions options) {
 		}
 	}
 
-	/**
-	 * Retrieves the {@link ActorGateway} for the JobManager. The ClusterClient is retrieved
-	 * from the provided {@link CommandLineOptions}.
-	 *
-	 * @param options CommandLineOptions specifying the JobManager URL
-	 * @return Gateway to the JobManager
-	 * @throws Exception
-	 */
-	protected ActorGateway getJobManagerGateway(CommandLineOptions options) throws Exception {
-		logAndSysout("Retrieving JobManager.");
-		return retrieveClient(options).getJobManagerGateway();
-	}
-
 	/**
 	 * Creates a {@link ClusterClient} object from the given command line options and other parameters.
-	 * @param options Command line options
+	 * @param customCommandLine custom command line to use to retrieve the client
+	 * @param commandLine command line to use
 	 * @param program The program for which to create the client.
 	 * @throws Exception
 	 */
 	protected ClusterClient createClient(
-			CommandLineOptions options,
+			CustomCommandLine<?> customCommandLine,
```

Comment:
Was it really necessary to change the signature? It seems that the behavior is otherwise the same. Why not leave the line 
```
CustomCommandLine<?> activeCommandLine = getActiveCustomCommandLine(options.getCommandLine());
```
inside this method.